src/components/PostDetails.tsx

```diff
@@ -1,117 +1,121 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { Error } from '../types/Error';
+
+type Props = {
+  postComments: Comment[],
+  selectedPost: Post,
+  handleDeleteComment: (commntId: number) => void,
+  setPostComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  isLoadingComments: boolean,
+  error: Error,
+  setError: React.Dispatch<React.SetStateAction<Error>>,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  postComments,
+  selectedPost,
+  handleDeleteComment,
+  setPostComments,
+  isLoadingComments,
+  error,
+  setError,
+}) => {
+  const [openCommentForm, setOpenCommentForm] = useState(false);
+
+  useEffect(() => {
+    setOpenCommentForm(false);
+  }, [selectedPost]);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${selectedPost.id}: ${selectedPost.title}`}
```

Comment:
Use destructuring to get `id`, `title` and others from `selectedPost` at the beginning of the component for cleaner code.